packages/core/src/com/communication.ts

```diff
@@ -118,9 +118,22 @@ export class Communication {
         const activeEnvironment = endPoint.environments.find(env => env.env === runtimeEnvironmentName)!;
         activeEnvironment.env = endPoint.env;
 
-        return activeEnvironment.envType === 'node'
-            ? this.connect(activeEnvironment as Environment<string, 'node'>)
-            : this.spawn(activeEnvironment);
+        if (activeEnvironment.envType === 'node') {
```

Comment:
maybe here the `activeEnvironment` can provide the "register" function implementation

Reply:
so we don't need a specific registerElectronEnvironment here

Reply:
by the way why registerElectronEnvironment is related to electron? isn't it register and listen?

Reply:
it's very similar to useWorker...